src/compiler/scala/tools/nsc/transform/patmat/MatchAnalysis.scala

```diff
@@ -516,11 +516,15 @@ trait MatchAnalysis extends MatchApproximation {
       //    - there are extractor calls (that we can't secretly/soundly) rewrite
       val start = if (Statistics.canEnable) Statistics.startTimer(patmatAnaExhaust) else null
       var backoff = false
+      val strict = settings.XstrictPatmatAnalysis.value
 
       val approx = new TreeMakersToPropsIgnoreNullChecks(prevBinder)
       val symbolicCases = approx.approximateMatch(cases, approx.onUnknown { tm =>
         approx.fullRewrite.applyOrElse[TreeMaker, Prop](tm, {
           case BodyTreeMaker(_, _) => True // irrelevant -- will be discarded by symbolCase later
+          case GuardTreeMaker(_) => if (strict) False else True
+          case ExtractorTreeMaker(extractor, _, _) if extractor.symbol.name != nme.unapplySeq =>
```

Comment:
@retronym I thought the guard here was because of SI-9567 (fixed in #4862, which seems to have been merged into 2.12.x at this point). However, removing it in order to get the correct `repeatedExtractor` warnings causes other tests to fail (e.g., run/patmatnew.scala:58, which is an exhaustive match on `List[Int]`).

Wondering if you think there’s a middle ground that can be achieved, or if the `repeatedExtractor` warnings are a bit too ambitious (at least for me).